runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/OpAdapter.java

```diff
@@ -84,8 +84,6 @@ public final void schedule(Scheduler<KeyedTimerData<K>> timerRegistry) {
 
   @Override
   public synchronized CompletionStage<Collection<OpMessage<OutT>>> apply(OpMessage<InT> message) {
```

Comment:
Do we still need the `synchronized` keyword across the methods given we have removed the shared state and push the responsibility to emitter?

Reply:
We still need here, e.g. to synchronize the underlying method calls. You pointed out the collectFuture() call in another comments. So keep this synchronized will avoid these complications.